openedx/core/djangoapps/schedules/resolvers.py

```diff
@@ -1,136 +1,313 @@
-                orgs that should be excluded from this send. All other orgs should be included.
+        Arguments:
+        schedule_date_field -- string field name to query on the User's Schedule model
+        current_datetime -- datetime that will be used as "right now" in the query
+        target_datetime -- datetime that the User's Schedule's schedule_date_field value should fall under
+        bin_num -- int for selecting the bin of Users whose id % num_bins == bin_num
+        num_bin -- int specifying the number of bins to separate the Users into (default: DEFAULT_NUM_BINS)
+        org_list -- list of course_org names (strings) that the returned Schedules must or must not be in (default: None)
+        exclude_orgs -- boolean indicating whether the returned Schedules should exclude (True) the course_orgs in org_list
+                        or strictly include (False) them (default: False)
+        order_by -- string for field to sort the resulting Schedules by
         """
-        try:
-            site_config = SiteConfiguration.objects.get(site_id=self.site.id)
-            org_list = site_config.get_value('course_org_filter')
-            exclude_orgs = False
-            if not org_list:
-                not_orgs = set()
-                for other_site_config in SiteConfiguration.objects.all():
-                    other = other_site_config.get_value('course_org_filter')
-                    if not isinstance(other, list):
-                        if other is not None:
-                            not_orgs.add(other)
-                    else:
-                        not_orgs.update(other)
-                org_list = list(not_orgs)
-                exclude_orgs = True
-            elif not isinstance(org_list, list):
-                org_list = [org_list]
-        except SiteConfiguration.DoesNotExist:
-            org_list = None
-            exclude_orgs = False
-        finally:
-            return exclude_orgs, org_list
+        target_day = _get_datetime_beginning_of_day(self.target_datetime)
+        schedule_day_equals_target_day_filter = {
+            'courseenrollment__schedule__{}__gte'.format(self.schedule_date_field): target_day,
+            'courseenrollment__schedule__{}__lt'.format(self.schedule_date_field): target_day + datetime.timedelta(days=1),
+        }
+        users = User.objects.filter(
+            courseenrollment__is_active=True,
+            **schedule_day_equals_target_day_filter
+        ).annotate(
+            id_mod=F('id') % self.num_bins
+        ).filter(
+            id_mod=self.bin_num
+        )
+
+        schedule_day_equals_target_day_filter = {
+            '{}__gte'.format(self.schedule_date_field): target_day,
+            '{}__lt'.format(self.schedule_date_field): target_day + datetime.timedelta(days=1),
+        }
+        schedules = Schedule.objects.select_related(
+            'enrollment__user__profile',
+            'enrollment__course',
```

Comment:
If it helps with keeping the query result compact - subclasses could specify which course attributes are really needed (instead of the entire course-overview entry).  For example, all the current subclasses need only the `display_name` and the `language`.

Reply:
Hmm.... Interesting thought. I'll look into it. I had discussed a pattern with Renzo for doing something similar in another piece of code.